src/Material.cpp

```diff
@@ -279,9 +279,9 @@ FP_PRECISION* Material::getDifHat() {
 
   if (_dif_hat == NULL) {
 
-    _dif_hat = new FP_PRECISION[4*_num_groups];
+    _dif_hat = new FP_PRECISION[6*_num_groups];
 
-    for (int e = 0; e < 4*_num_groups; e++)
+    for (int e = 0; e < 6*_num_groups; e++)
```

Comment:
What about 2D problems - does changing from 4 to 6 affect them? I assume not, but why?